docs/Lexicon.md

```diff
@@ -484,6 +484,39 @@ See [mandatory passes](#mandatory-passes--mandatory-optimizations).
 An implicit representation change that occurs when a value is used with
 a different [abstraction pattern](#abstraction-pattern) from its current representation.
 
+## realization
```

Comment:
cc @mikeash, could you double-check if the description in this commit for realization and `@_objc_non_lazy_realization` is alright? It's mostly based on our conversation, but just in case I messed something up.

Reply:
Made a few small comments, overall it's good.